packages/pigeon/lib/kotlin_generator.dart

```diff
@@ -839,6 +1201,624 @@ class KotlinGenerator extends StructuredGenerator<KotlinOptions> {
       });
     });
   }
+
+  void _writeProxyApiRegistrar(
+    Indent indent, {
+    required Iterable<AstProxyApi> allProxyApis,
+  }) {
+    const String registrarName = '${classNamePrefix}ProxyApiRegistrar';
+    const String instanceManagerApiName = '${instanceManagerClassName}Api';
+
+    indent.format(
+      '/**\n'
+      ' * Provides implementations for each ProxyApi implementation and provides access to resources\n'
+      ' * needed by any implementation.\n'
+      ' */',
+    );
+    indent.writeScoped(
+      'abstract class $registrarName(val binaryMessenger: BinaryMessenger) {',
+      '}',
+      () {
+        indent.writeln('val instanceManager: $instanceManagerClassName');
+        indent.format(
+          'private var _codec: StandardMessageCodec? = null\n'
+          'val codec: StandardMessageCodec\n'
+          '  get() {\n'
+          '    if (_codec == null) {\n '
+          '      _codec = PigeonProxyApiBaseCodec(this)\n'
+          '    }\n'
+          '    return _codec!!\n'
+          '  }\n',
+        );
+        indent.format(
+          'init {\n'
```

Comment:
`'''` to improve legibility?

Reply:
I've tried to use `'''` with format, but the code that comes out (before running the formatter) includes the empty spaces unless you do something like this: https://github.com/flutter/packages/blob/main/packages/pigeon/lib/dart_generator.dart#L414

I updated `Indent.format` with a flag that can trim the indentation: https://github.com/flutter/packages/pull/6371/commits/78128b8dee3001f49b8f4bedaa5ae5731e9e71c6. Let me know if you believe there is a cleaner way.

cc @tarrinneal